### PR TITLE
Add instrumentation test [run instrumentation test]

### LINKED ISSUE #303: Add just launch Instrumentation Test
## Kind (Required)

- Improvement

 ## Overview (Required)

- Can you add just a launch test?

 ## Links	

-

### PR DESCRIPTION
**Pull request by hkusu**
## Issue
- close #303 

## Overview (Required)
- Added **Espresso** based instrumentation test.
- Can be run as
  - `./gradlew android-base:connectedDebugAndroidTest`

## Links
-

## Other
- Add GitHub Actions for operation check
  - May not be stable..
- test results
  - https://github.com/DroidKaigi/conference-app-2020/runs/394797680

スクリーンショット 2020-01-17 17 47 28

### CODE REVIEW

File: android-base/build.gradle [hunk 1]
```diff
@@ -17,6 +17,7 @@ android {
         versionCode Versions.androidVersionCode
         versionName Versions.androidVersionName
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunnerArguments clearPackageData: 'true'
```
Comment (hkusu) [inline android-base/build.gradle]: for Android Test Orchestrator

File: android-base/build.gradle [hunk 2]
```diff
@@ -68,6 +69,10 @@ android {
             ':feature:preference',
             ':feature:contributor'
     ]
+    testOptions {
+        execution 'ANDROIDX_TEST_ORCHESTRATOR'
```
Comment (hkusu) [inline android-base/build.gradle]: for Android Test Orchestrator

File: android-base/build.gradle [hunk 3]
```diff
@@ -68,6 +69,10 @@ android {
...
+        animationsDisabled = true
```
Comment (hkusu) [inline android-base/build.gradle]: Disable animation. but, does not work depending on API level..

File: corecomponent/androidtestcomponent/build.gradle [hunk 4]
```diff
@@ -14,7 +14,6 @@ dependencies {
     implementation Dep.Kotlin.coroutines
 
     api Dep.Test.archCore
-    api Dep.Test.testCore
```
Comment (hkusu) [inline corecomponent/androidtestcomponent/build.gradle]: Deleted because it was unnecessary

File: buildSrc/src/main/java/dependencies/Dep.kt [hunk 5]
```diff
@@ -19,8 +19,10 @@ object Dep {
     object Test {
         val junit = "junit:junit:4.12"
         val testRunner = "androidx.test:runner:1.3.0-alpha02"
-        val testCore = "androidx.test:core:1.1.0"
-        val androidJunit4 = "androidx.test.ext.junit:1.1.0"
+        val testRules = "androidx.test:rules:1.3.0-alpha02"
+        val testCoreKtx = "androidx.test:core-ktx:1.2.1-alpha02"
+        val androidJunit4Ktx = "androidx.test.ext:junit-ktx:1.1.2-alpha02"
```
Comment (hkusu) [inline buildSrc/src/main/java/dependencies/Dep.kt]: Introduced `-ktx`

File: .github/workflows/on_pull_request.yml [hunk 6]
```diff
@@ -163,3 +163,35 @@ jobs:
...
+            ${{ runner.os }}-m2-
+      - name: Run instrumentation test
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: ${{ matrix.api-level }}
+          target: google_apis
+          arch: x86
+          profile: pixel
+          disable-animations: true
+          script: ./gradlew android-base:connectedDebugAndroidTest
```
Comment (hkusu) [inline .github/workflows/on_pull_request.yml]: As I commented, this is a temporary one to check the test. You can delete it.

File: .github/workflows/on_pull_request.yml [hunk 7]
```diff
@@ -163,3 +163,35 @@ jobs:
       # - run: bundle exec danger --dangerfile=".ci/danger/apk_comparison.Dangerfile" --danger_id='apk_comparison' --remove-previous-comments
       #   env:
       #     DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  # TODO This is a temporary one to check the test. You can delete it.
+  # Emurator's test is not stable now. Sometimes fail..
+  instrumentation-test:
+    runs-on: macOS-latest
+    if: contains(github.event.pull_request.title, '[run instrumentation test]') == true
```
Comment (hkusu) [inline .github/workflows/on_pull_request.yml]: Only works when `[run instrumentation test]` is in the title of the pull request.

**Comment by takahirom** [comment]
👀

File: .github/workflows/on_pull_request.yml [hunk 8]
```diff
@@ -163,3 +163,35 @@ jobs:
...
+    strategy:
+      matrix:
```
Comment (takahirom) [inline .github/workflows/on_pull_request.yml]: 👍

**Pull request merged**